mungegithub/mungers/triage-filer.go

```diff
@@ -0,0 +1,582 @@
+//then syncs the top issues to github with the IssueCreator.
+func (filer *TriageFiler) FileIssues() error {
+	rawjson, err := mungerutil.FetchWebFileContents(filer.clusterDataURL)
+	if err != nil {
+		return err
+	}
+	clusters := filer.parseClusters(rawjson)
+	topclusters := topClusters(clusters, filer.topClustersCount)
+	for _, clust := range topclusters {
+		filer.creator.Sync(clust)
+	}
+	return nil
+}
+
+//AddFlags will add any requested flags to the cobra `cmd`.
+func (filer *TriageFiler) AddFlags(cmd *cobra.Command, config *github.Config) {
+	cmd.Flags().StringVar(&filer.clusterDataURL, "clusterdataurl", "https://storage.googleapis.com/k8s-gubernator/triage/failure_data.json", "URL pointing to JSON file from triage page that describes issue clusters.")
+	cmd.Flags().IntVar(&filer.topJobsCount, "topjobscount", 3, "The maximum number of jobs to show in an issue under 'Top failing jobs'.")
+	cmd.Flags().IntVar(&filer.topTestsCount, "toptestscount", 3, "The maximum number of tests to show in an issue under 'Top failing tests'.")
+	cmd.Flags().IntVar(&filer.topClustersCount, "triagesynccount", 10, "The number of clusters to sync issues for on github.")
+	cmd.Flags().StringVar(&filer.triageURL, "triageURL", "https://go.k8s.io/triage", "The url prefix to add before cluster hash to create a link to the triage page for a specific cluster.")
+	cmd.Flags().IntVar(&filer.windowDays, "triagewindowdays", 5, "The size of the sliding time window (in days) that is used to determine which failures to consider.")
+	cmd.Flags().IntVar(&filer.syncFreqMins, "triagesyncfreq", 60, "The frequency at which to run the TriageFiler in minutes.")
+}
+
+//Munge is unused by the TriageFiler.
+func (filer *TriageFiler) Munge(obj *github.MungeObject) {}
+
+//topClusters gets the 'count' most important clusters from a slice of clusters based on number of build failures.
+func topClusters(clusters []*Cluster, count int) []*Cluster {
+	iclusters := make([]interface{}, len(clusters))
+	for i := 0; i < len(iclusters); i++ {
+		iclusters[i] = clusters[i]
+	}
+
+	sorter := &slicesorter{slice: iclusters,
+		less: func(a, b int) bool { return iclusters[a].(*Cluster).totalBuilds > iclusters[b].(*Cluster).totalBuilds },
+	}
+
+	sort.Sort(sorter)
+
+	if len(iclusters) < count {
+		count = len(iclusters)
+	}
+	result := make([]*Cluster, count)
+	for i := 0; i < count; i++ {
+		result[i] = iclusters[i].(*Cluster)
+	}
+	return result
+}
+
+//parseClusters generates a 'Cluster' struct for every cluster in the provided JSON that has
+//data that falls within the time window. Failures from before the time window
+//are ignored.
+func (filer *TriageFiler) parseClusters(jsonIn []byte) []*Cluster {
+	var clusters []*Cluster
+
+	//first parse the "clustered" and "builds" keys from the top level
+	topLevelDict := make(map[string]json.RawMessage)
+	err := json.Unmarshal(jsonIn, &topLevelDict)
+	if err != nil {
+		glog.Fatalln("ERROR: ", err)
```

Comment:
Don't use Fatal in production code unless it's *actually* unrecoverable. Log an error and return nil, or make parseClusters return ([]*Cluster, err).